ca_bc_victoria/people.py

```diff
@@ -1,7 +1,47 @@
-from utils import CSVScraper
+from utils import CanadianPerson as Person
+from utils import CanadianScraper
 
+COUNCIL_PAGE = "https://www.victoria.ca/city-government/mayor-council/members-council"
+MAYOR_PAGE = "https://www.victoria.ca/city-government/mayor-council/mayor-marianne-alto"  # will probably break if the mayor changes
 
-class VictoriaPersonScraper(CSVScraper):
-    # http://opendata.victoria.ca/datasets/councillor-contact-information-2018-2022
-    csv_url = "https://opendata.arcgis.com/datasets/d524b5f7e64b40a5ae118dde1852fc22_0.csv?outSR=4326"
-    many_posts_per_area = True
+
+class VictoriaPersonScraper(CanadianScraper):
+    def scrape(self):
+        seat_number = 1
+        page = self.lxmlize(COUNCIL_PAGE)
+
+        councillors = page.xpath(
+            '//article[@class="node node--type-councillor node--view-mode-list-item list-item flex fd-r fd-c-mq-m g-32"]'
+        )
+
+        for councillor in councillors:
+            role, name = councillor.xpath(".//h3/a/span")[0].text_content().split(" ", 1)
+            photo = councillor.xpath(".//img/@src")[0]
+            email = self.get_email(councillor)
+            phone = self.get_phone(councillor)
+            url = councillor.xpath(".//h3/a/@href")[0]
+
+            district = "Victoria (seat {})".format(seat_number)
+            seat_number += 1
+
+            p = Person(primary_org="legislature", name=name, district=district, role=role)
+            p.image = photo
+            p.add_contact("email", email)
+            p.add_contact("voice", phone, "legislature")
+            p.add_source(COUNCIL_PAGE)
+            p.add_source(url)
+
+            yield p
+
+        page = self.lxmlize(MAYOR_PAGE)
```

Comment:
@rafe-murray what James is suggesting is to do something like this to avoid the hard coding of the mayors page

```python
menu_links = page.xpath('//ul[@class="menu menu--level-0"]/li/ul/li/span/a@href')
for link in menu_links:
  if "mayor" not in link:
    continue
  MAYOR_PAGE = link
```

Reply:
Got it, that makes sense. Thanks Seamus!